pkg/validator/validator.go

```diff
@@ -14,10 +14,22 @@ type Validator interface {
 	Validate(group unmarshaler.RuleGroup, rule rulefmt.Rule, prometheusClient *prometheus.Client) []error
 }
 
+const (
+	matchAnythingRegexp = ".*"
+	emptyRegexp         = ""
+)
+
 func anchorRegexp(regexpString string) string {
 	return fmt.Sprintf("^%s$", regexpString)
 }
 
-func compileAnchoredRegexp(regexpString string) (*regexp.Regexp, error) {
+func compileAnchoredRegexpWithDefault(regexpString, defaultValue string) (*regexp.Regexp, error) {
```

Comment:
Idea: got me thinking that it could be a type with custom `UnmarshalYAML` something like `RegexpEmptyDefault` and `RegexpWildcardDefault` and it would save lines with calling and error handling this function :thinking:

Reply:
Good point, makes sense. :)

Refactored in 1fef8919d2a85ef182edda550b1bf835e0b296b3. ptal

Reply:
I've changed it a bit more to include regexp compile within the unmarshalling itself.